src/Beman/Optional26/tests/optional_constexpr.t.cpp

```diff
@@ -0,0 +1,1261 @@
+    constexpr beman::optional26::optional<NoDefault> v2{i};
+    std::ignore = v1;
+    std::ignore = v2;
+}
+
+consteval bool testConstexprAssignmentValue() {
+    bool retval = true;
+    beman::optional26::optional<int> o1 = 42;
+    beman::optional26::optional<int> o2 = 12;
+    beman::optional26::optional<int> o3;
+
+    o1 = o1;
+    retval &= (*o1 == 42);
+
+    o1 = o2;
+    retval &= (*o1 == 12);
+
+    o1 = o3;
+    retval &= (!o1);
+
+    o1 = 42;
+    retval &= (*o1 == 42);
+
+    o1 = beman::optional26::nullopt;
+    retval &= (!o1);
+
+    o1 = std::move(o2);
+    retval &= (*o1 == 12);
+
+    beman::optional26::optional<short> o4 = 42;
+
+    o1 = o4;
+    retval &= (*o1 == 42);
+
+    o1 = std::move(o4);
+    retval &= (*o1 == 42);
+
+    /*
+      template <class U = T>
+      constexpr optional& operator=(U&& u)
+    */
+    short s = 54;
+    o1      = s;
+    retval &= (*o1 == 54);
+
+    struct not_trivial_copy_assignable {
+        int i_;
+        constexpr not_trivial_copy_assignable(int i) : i_(i) {}
+        constexpr not_trivial_copy_assignable(const not_trivial_copy_assignable&) = default;
+        constexpr not_trivial_copy_assignable& operator=(const not_trivial_copy_assignable& rhs) {
+            i_ = rhs.i_;
+            return *this;
+        }
+    };
+
+    /*
+      optional& operator=(const optional& rhs)
+      requires std::is_copy_constructible_v<T> && std::is_copy_assignable_v<T> &&
+      (!std::is_trivially_copy_assignable_v<T>)
+    */
+    beman::optional26::optional<not_trivial_copy_assignable> o5{5};
+    beman::optional26::optional<not_trivial_copy_assignable> o6;
+    o6 = o5;
+    retval &= (o5->i_ == 5);
+    return retval;
+}
+
+static_assert(testConstexprAssignmentValue());
+
+
+struct takes_init_and_variadic {
+    int     v0;
+    std::tuple<int, int> t;
+    template <class... Args>
+    constexpr takes_init_and_variadic(std::initializer_list<int>& l, Args&&... args) : v0{*std::begin(l)}, t(std::forward<Args>(args)...) {}
+};
+
+consteval bool testConstexprInPlace() {
+    bool retval = true;
+    constexpr beman::optional26::optional<int> o1{beman::optional26::in_place};
+    constexpr beman::optional26::optional<int> o2(beman::optional26::in_place);
+    retval &= (bool(o1));
+    retval &= (o1 == 0);
+    retval &= (bool(o2));
+    retval &= (o2 == 0);
+
+    constexpr beman::optional26::optional<int> o3(beman::optional26::in_place, 42);
+    retval &= (o3 == 42);
+
+    constexpr beman::optional26::optional<std::tuple<int, int>> o4(beman::optional26::in_place, 0, 1);
+    retval &= (bool(o4));
+    retval &= (std::get<0>(*o4) == 0);
+    retval &= (std::get<1>(*o4) == 1);
+
+    constexpr beman::optional26::optional<takes_init_and_variadic> o6(beman::optional26::in_place, {0, 1}, 2, 3);
+    retval &= (o6->v0 == 0);
+    retval &= (std::get<0>(o6->t) == 2);
+    retval &= (std::get<1>(o6->t) == 3);
+    return retval;
+}
+
+auto consteval constify(auto expr) { return (expr); }
```

Comment:
Please add docs for utility functions. Any reason not to move this in the test_type header?